### PR TITLE
ci: Add release checks and jobs

### PR DESCRIPTION
**Pull request by jrobsonchase**
Alright, I'm reasonably happy with this now. Semver checks now happen before PR merge, and once it hits master, it'll always try to publish, and if the publish succeeds, it'll add a git tag for us.

So all we have to do to publish is make a PR with a version bump!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,12 +13,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: cachix/install-nix-action@v18
-      - run: |
-          curl -v --retry 5 --retry-all-errors -sfL https://direnv.net/install.sh \
-            | sed 's/curl/curl -v --retry 5 --retry-all-errors --header "authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"/g' \
-            | bash
-      - run: direnv allow
-      - run: direnv export gha >> "$GITHUB_ENV"
+      - uses: jrobsonchase/direnv-action@v0.6
```
Comment (bobzilladev) [inline .github/workflows/ci.yml:16]: Nice, that's a lot cleaner.

One thing I'll bring up: It doesn't seem to be causing issues right now, but saw when going through this before that the release download url is also on github, so the second curl might also benefit from the token. That happened when I was forcing a curl `alias` into the bash, but not with the `sed` or now `action` methods. Can probably wait and see if we ever see that fail, but may be something that needs to be done someday.

https://github.com/jrobsonchase/direnv-action/blob/51cf7cfbd706817008345510e72068d766df1825/install.sh#L91

[burn after reading]

**Review by bobzilladev — APPROVED** [review]
lgtm. stealing the repository check for ngrok-js.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jrobsonchase) [reply .github/workflows/ci.yml:16]: Ah, interesting. I've been fiddling with this action all morning to try to get it to cache nix stuff betterer and haven't hit any issues with that one yet. Maybe it's an api.github.com vs plain github.com thing?

Comment (bobzilladev) [reply .github/workflows/ci.yml:16]: ah, could be

**Pull request merged**